src/theory/strings/regexp_operation.cpp

```diff
@@ -655,8 +679,8 @@ Node RegExpOpr::derivativeSingle( Node r, CVC4::String c ) {
         break;
       }
       default: {
-        Trace("strings-error") << "Unsupported term: " << mkString( r ) << " in derivative of RegExp." << std::endl;
-        Unreachable();
+        Assert(!isRegExpKind(k));
+        break;
```

Comment:
I think that we should return the null node here and then bail in `intersectInternal` if we get a null node, otherwise the intersection may be empty even though it should not be.

Reply:
I'm going to change this back to `Unreachable();`. This shouldn't be called on regular expressions with subterms that are not regular expression kinds.